packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -562,6 +649,33 @@ class TextPainter {
     _layoutTemplate = null; // Shouldn't really matter, but for strict correctness...
   }
 
+  static double _computePaintOffsetFraction(TextAlign textAlign, TextDirection textDirection) {
+    switch (textAlign) {
+      case TextAlign.left:
+        return 0;
+      case TextAlign.right:
+        return 1;
+      case TextAlign.center:
+        return 0.5;
+      case TextAlign.start:
+      // eot is also considered a hard break.
```

Comment:
not sure what this comment means

Reply:
removed. I thought if the text doesn't contain hard breaks the extra width will be distributed among the whitespaces in that line, but turns out the end of line is counted as a hard break so when maxWidth > maxIntrinsicWidth it's the same as TextAlign.start.